src/audio/dai.c

```diff
@@ -157,6 +157,20 @@ static void dai_dma_cb(void *data, uint32_t type, struct dma_sg_elem *next)
 			trace_dai_error("xro");
 			comp_overrun(dev, dma_buffer, dd->period_bytes, 0);
 		}
+
+		/* when capture paused, the buffer should be reset,
+		 * the dma pointer should sync with buffer pointer.
+		 */
+		if (dev->state != COMP_STATE_ACTIVE) {
+			struct dma_chan_status status;
+
+			buffer_reset_pos(dma_buffer);
+			dma_status(dd->dma, dd->chan,
+				   &status, SOF_IPC_STREAM_CAPTURE);
+
+			if (status.w_pos != (uint32_t)dma_buffer->w_ptr)
```

Comment:
Why do you release the DMA in the callback if stream is not active ?

Reply:
When the pipeline is paused, this is last time we reach here before release the pipeline.
over here, we reset the buffer pointer, call the dma_release() to adjust the dma pointer.
then ready for the release command coming.

Reply:
Consider a DAI capture buffer B with 2 periods p0, p1. At release the downstream pipeline will consume 0s from p0 whilst the DMA is filling p1. Once p1 is full DMA will IRQ and then the DMA will fill p0 and pipeline will consume valid data from p1 (and so on...)

Reply:
Actually we did not schedule the capture pipeline when we do the start/release. the capture's pipeline is scheduled when the DMA finish filling one period buffer. At the beginning stage, the downstream pipeline is not working, actually only the dai DMA is doing the data filling.
So we have to reset the buffer pointer and DMA pointer over here.